sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCScanBuilder.scala

```diff
@@ -36,6 +37,8 @@ case class JDBCScanBuilder(
 
   private var prunedSchema = schema
 
+  var partition = Array.empty[Partition]
```

Comment:
How about to avoid `var`, and add `partition` to the case class `JDBCScanBuilder`, and use `.copy`?